client/client.go

```diff
@@ -368,6 +368,11 @@ func (c *Client) SetDebug(w io.Writer) {
 	c.conn.SetDebug(w)
 }
 
+// Conn returns the client's imap.Conn.
+func (c *Client) Conn() *imap.Conn {
```

Comment:
I would also prefer the Timeout field. It's not that complicated as you could just add:
```go
if c.Timeout > 0 {
    c.conn.SetDeadline(time.Now().Add(c.Timeout))
}
```
in `executeCommand`.